dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/user/DefaultUserService.java

```diff
@@ -618,49 +619,44 @@ public boolean isAccountExpired(User user) {
 
   @Override
   @Transactional(readOnly = true)
-  public List<ErrorReport> validateUser(User user, User currentUser) {
+  public List<ErrorReport> validateUserCreateOrUpdate(User user, User currentUser) {
+
     List<ErrorReport> errors = new ArrayList<>();
 
     if (currentUser == null || user == null) {
       return errors;
     }
 
-    // Validate user role
+    // Validate if the current user can create/modify users with the ALL authority
+    User userToChange = userStore.get(user.getId());
+    if (!currentUser.isSuper() && userToChange != null && userToChange.getUserRoles() != null) {
+      Set<UserRole> roles = userToChange.getUserRoles();
+      for (UserRole role : roles) {
+        Set<String> authorities = role.getAuthorities();
+        if (authorities.contains(Authorities.ALL.name())) {
```

Comment:
There's a helper method in `User` to get all authorities directly without having to get the roles first - `getAllAuthorities`
Should it break out of the loop if it finds the `ALL` authority? There's probably no point checking any more if we know it has the `ALL` authority.